### PR TITLE
Always use in-memory database in tests

### PR DESCRIPTION
**Pull request by laevandus**
### 🔗 Issue Links

Resolves [PBE-4302](https://stream-io.atlassian.net/browse/PBE-4302)

### 🎯 Goal

* Use in-memory database for tests except performance tests (faster, less errors with SQLite).
* Fix skipped channel list tests
* Fix unstable tests what appeared after background mapping is the default (we need to wait delegates since updates have a delay)

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [ ] Changelog is updated with new localization keys
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

[PBE-4302]: https://stream-io.atlassian.net/browse/PBE-4302?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: TestTools/StreamChatTestTools/SpyPattern/Spy/DatabaseContainer_Spy.swift [hunk 1]
```diff
@@ -31,8 +31,7 @@ public final class DatabaseContainer_Spy: DatabaseContainer, Spy {
     private(set) var sessionMock: DatabaseSession_Mock?
 
     public convenience init(localCachingSettings: ChatClientConfig.LocalCaching? = nil) {
-        self.init(kind: .onDisk(databaseFileURL: .newTemporaryFileURL()), localCachingSettings: localCachingSettings)
```
Comment (nuno-vieira) [inline TestTools/StreamChatTestTools/SpyPattern/Spy/DatabaseContainer_Spy.swift:34]: Really? 🫨 I thought we were using in-memory everywhere 😅

Comment (laevandus) [reply TestTools/StreamChatTestTools/SpyPattern/Spy/DatabaseContainer_Spy.swift:34]: This init is used a lot. Don't think there is real requirement for creating the actual DB file.

File: Tests/StreamChatTests/APIClient/Endpoints/Payloads/ChannelListPayload_Tests.swift [hunk 2]
```diff
@@ -65,7 +65,7 @@ final class ChannelListPayload_Tests: XCTestCase {
     }
 
     func test_hugeChannelListQuery_save_DB_empty() throws {
-        throw XCTSkip("https://github.com/GetStream/ios-issues-tracking/issues/848")
+        // throw XCTSkip("https://github.com/GetStream/ios-issues-tracking/issues/848")
```
Comment (laevandus) [inline Tests/StreamChatTests/APIClient/Endpoints/Payloads/ChannelListPayload_Tests.swift:68]: Just testing currently

**Review by testableapple — APPROVED** [review]
You're legend! 🤠

File: TestTools/StreamChatTestTools/SpyPattern/Spy/DatabaseContainer_Spy.swift (hunk 1 above)
Comment (nuno-vieira) [reply TestTools/StreamChatTestTools/SpyPattern/Spy/DatabaseContainer_Spy.swift:34]: Maybe for performance testing it is more realistic, but not a big deal I guess 🤔

File: Tests/StreamChatTests/Controllers/ChannelController/ChannelController_Tests.swift [hunk 3]
```diff
@@ -1734,7 +1734,7 @@ final class ChannelController_Tests: XCTestCase {
         // Unlike new DM ChannelController, this ChannelController knows it's final `cid` so it should be able to fetch initial values
         // from DB, without the `synchronize` call
         // Assert that initial reported values are correct
-        XCTAssertEqual(controller.channel?.cid, dummyChannel.channel.cid)
+        AssertAsync.willBeEqual(controller.channel?.cid, dummyChannel.channel.cid)
```
Comment (laevandus) [inline Tests/StreamChatTests/Controllers/ChannelController/ChannelController_Tests.swift:1737]: All of these others are related to the fact that background mapping was turned on and now controllers delay their updates. Not sure why timing issues appeared now, maybe the DB change changed timings.

File: Tests/StreamChatTests/Controllers/MessageController/MessageController_Tests.swift [hunk 4]
```diff
@@ -410,7 +410,7 @@ final class MessageController_Tests: XCTestCase {
 
         // Check the order of replies is correct
         let bottomToTopIds = [reply1, reply2].sorted { $0.createdAt < $1.createdAt }.map(\.id)
-        XCTAssertEqual(controller.replies.map(\.id), bottomToTopIds)
+        AssertAsync.willBeEqual(controller.replies.map(\.id), bottomToTopIds)
```
Comment (martinmitrevski) [inline Tests/StreamChatTests/Controllers/MessageController/MessageController_Tests.swift:413]: we're actually trying to get rid of `AssertAsync`, best to use raw expectations here

File: TestTools/StreamChatTestTools/SpyPattern/Spy/DatabaseContainer_Spy.swift (hunk 1 above)
Comment (martinmitrevski) [reply TestTools/StreamChatTestTools/SpyPattern/Spy/DatabaseContainer_Spy.swift:34]: this is a good point actually, maybe we can keep it for the performance checks?

File: Tests/StreamChatTests/Controllers/MessageController/MessageController_Tests.swift (hunk 4 above)
Comment (laevandus) [reply Tests/StreamChatTests/Controllers/MessageController/MessageController_Tests.swift:413]: I'll try to change these

File: TestTools/StreamChatTestTools/SpyPattern/Spy/DatabaseContainer_Spy.swift (hunk 1 above)
Comment (laevandus) [reply TestTools/StreamChatTestTools/SpyPattern/Spy/DatabaseContainer_Spy.swift:34]: I'll change it a bit so we use a file for performance tests

Comment (laevandus) [reply TestTools/StreamChatTestTools/SpyPattern/Spy/DatabaseContainer_Spy.swift:34]: I changed it to onDisk and these huge payload tests stopped working immediately. Don't understand still why it is so.

**Review by martinmitrevski — APPROVED** [review]
LGTM ✅  Please check the failing test before merging

File: Tests/StreamChatTests/APIClient/Endpoints/Payloads/ChannelListPayload_Tests.swift [hunk 5]
```diff
@@ -7,6 +7,17 @@
 import XCTest
 
 final class ChannelListPayload_Tests: XCTestCase {
+    private var database: DatabaseContainer_Spy!
+    
+    override func setUpWithError() throws {
+        // Clean up any db files left behind
+        FileManager.removeAllTemporaryFiles()
```
Comment (laevandus) [inline Tests/StreamChatTests/APIClient/Endpoints/Payloads/ChannelListPayload_Tests.swift:14]: Seems like the simulator had a lot of old DB files left behind

**Pull request merged**